src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java

```diff
@@ -51,4 +51,19 @@ public EmployeeDto getEmployeeByUserName(String username) {
     BeanUtils.copyProperties(employee.get(), employeeDto);
     return employeeDto;
   }
+
+  @Override
+  public EmployeeDto updateEmployeeEntity(EmployeeDto employeeDto) {
+    Optional<Employee> employee = employeeRepository.findById(employeeDto.getId());
+    if (!employee.isPresent()) {
+      ExceptionUtils
+          .throwEntityNotFoundException(
+              String
+                  .format("Employee with Id %d doesn't exists.", employeeDto.getId()));
+    }
+
```

Comment:
Don't allow username and id of employee to get updated...throw bad request exception is someone is trying to update them..and take special care..that they don't get updated.